src/adam_core/propagator/adam_assist.py

```diff
@@ -517,10 +523,127 @@ def _detect_impacts(
                 variant_id=[],
             )
         return results, earth_impacts
-
+    
+    def _add_light_time(
+        self, 
+        orbits,
+        observers,
+        lt_tol: float = 1e-10,
+    ):  
+        orbits_aberrated = orbits.empty()
+        lts = np.zeros(len(orbits))
+        for i, (orbit, observer) in enumerate(zip(orbits, observers)):
+            lt0 = 0
+            dlt = 1
+            while dlt > lt_tol:
```

Comment:
I don't see a any handler for max-attempts. Is this loop guaranteed to converge? (I honestly don't know).